go-controller/pkg/cni/cniserver.go

```diff
@@ -158,7 +155,20 @@ func (s *Server) handleCNIRequest(w http.ResponseWriter, r *http.Request) {
 			klog.Warningf("Error writing %s HTTP response: %v", req.Command, err)
 		}
 	}
+}
+
+func (s *Server) handleCNIMetrics(w http.ResponseWriter, r *http.Request) {
+	var cm CNIRequestMetrics
 
-	metrics.MetricCNIRequestDuration.WithLabelValues(string(req.Command), hasErr).Observe(
-		time.Since(startTime).Seconds())
+	b, _ := ioutil.ReadAll(r.Body)
+	if err := json.Unmarshal(b, &cm); err != nil {
+		http.Error(w, fmt.Sprintf("%v", err), http.StatusBadRequest)
```

Comment:
The caller is currently ignoring the return value of the request, which seems correct, since it's hard to usefully log anything from a CNI plugin anyway. So we should log the error here in the server. (And possibly not even bother returning it to the plugin?)

Reply:
@danwinship I am working on a PR to log error messages from the CNIshim.

Right now, I have to grok through `kubelet` journal to find out what went wrong.

The idea here is to update the 10-ovn-kubernetes.conf with LogFile and LogLevel fields. This will be read by the ovn-k8s-cni-overlay binary at the beginning and setup the `klog` to dump the messages to that file.

I will capture this error message in that log file.

I have this PR almost ready, but I am facing some test failure with Multus + OVN with my changes.

Reply:
@danwinship on second thoughts, I agree with what you say :-). So, I have updated the PR to log a warning in the CNIServer itself.